app/javascript/mastodon/api.ts

```diff
@@ -62,12 +62,15 @@ export default function api(withAuthorization = true) {
 export async function apiRequest<ApiResponse = unknown>(
   method: Method,
   url: string,
-  params?: Record<string, unknown>,
+  args: {
+    params?: Record<string, unknown>;
+    data?: Record<string, unknown>;
+  } = {},
 ) {
   const { data } = await api().request<ApiResponse>({
     method,
     url: '/api/' + url,
-    data: params,
+    ...args,
   });
 
   return data;
```

Comment:
Looks ok to me, but I wonder if it really makes sense to wrap them in an `args` parameter? Also it may make sense to export `post`, `get`, `put` and `delete` aliases/helpers like axios does.

Reply:
An alternative is to use either `params` or `data`, depending on the HTTP verb. I think it would make more sense than having 4 different functions, and does not make the code more complex.

The big questions here is: do we have any request that might need both `params` and `data`? Are `POST` and `PUT` requests always using form data, and other requests always using URL parameters?

Reply:
I don't think we currently ever mix query strings and form data, and I'm not sure there is a case where that would ever make a lot of sense.

I think I'd prefer axios' approach of having one complete method that can do all, including mixing query strings and form data, and have shorthand aliases that fit common use cases, but I'm not opposed to another approach.